File: demo/OpenHospital/baseline/backend/schemas/doctor.py
"""Doctor schemas."""

from typing import List, Optional, Literal
from pydantic import BaseModel


class DoctorSummary(BaseModel):
    """Doctor summary for list view."""
    id: str
    name: str
    department: str
    specialties: List[str]
    current_status: Literal["idle", "consulting"] = "idle"
    patient_count: int = 0


class Doctor(BaseModel):
    """Full doctor profile."""
    id: str
    name: str
    department: str
    specialties: List[str]
    consultation_room: str
    current_status: Optional[str] = "idle"
    current_patients: Optional[List[dict]] = None


class DoctorStatistics(BaseModel):
    """Doctor statistics."""
    total_patients: int = 0
    diagnosis_accuracy: float = 0.0
    completed_consultations: int = 0

File: demo/OpenHospital/baseline/frontend/src/types/doctor.ts
export interface Doctor {
  id: string                     // e.g., "Doctor_Cardiology_001"
  name: string                   // e.g., "Dr. James Wilson"
  department: string             // e.g., "Cardiology Department"
  specialties: string[]          // e.g., ["Heart Failure", "Hypertension"]
  consultation_room: string      // e.g., "consultation_cardiology"
  initial_location: string       // e.g., "consultation_cardiology"
  current_status?: DoctorStatus
}

export type DoctorStatus = 'idle' | 'consulting'

export interface DoctorSummary {
  id: string
  name: string
  department: string
  specialties: string[]
  current_status: DoctorStatus
  patient_count: number
}

export interface DoctorDetail extends Omit<Doctor, 'current_status'> {
  current_status: DoctorStatus
  current_patients: PatientSummary[]
}

export interface DoctorStatistics {
  total_patients: number
  diagnosis_accuracy: number
  completed_consultations: number
}
import type { PatientSummary } from './patient'
